### PR TITLE
Expose Slab Correction in Electrostatics Wrappers

### PR DESCRIPTION
**Pull request by atulcthakur**
# ALCHEMI Toolkit Pull Request

## Description

Adds `slab_correction` support to Toolkit Ewald/PME wrappers, passing slab pbc geometry into ops only when requested. Refactors Ewald to use the full `ewald_summation` wrapper so slab composition and hybrid charge-gradient behavior stay consistent with ops.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Added `slab_correction` constructor arg to Ewald and PME wrappers.
- Passed `data.pbc` through only when slab correction is enabled.
- Routed Ewald through `ewald_summation` and PME through `particle_mesh_ewald` with slab args.
- Preserved existing stress/virial error behavior.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [ ] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit/pull/103/commits/1e4e2568c24261d245dfd05128b5fdd0920d3579

File: nvalchemi/models/ewald.py [hunk 1]
```diff
@@ -94,6 +94,9 @@ class EwaldModelWrapper(nn.Module, BaseModelMixin):
     coulomb_constant : float, optional
         Coulomb prefactor :math:`k_e` in eV·Å/e².
         Defaults to ``14.3996`` (standard value for Å/e/eV unit system).
+    slab_correction : bool, optional
```
Comment (laserkelvin) [inline nvalchemi/models/ewald.py:97]: Could you expand this docstring with the slab contract exercised in `test/models/test_slab.py:225-243` and `test/models/test_slab.py:254-273`? Specifically: `data.pbc` is a bool tensor shaped `(B, 3)`, the non-periodic slab axis is `False` such as `[True, True, False]`, fully periodic masks are no-ops, and mixed slab/3D batches are supported.

File: nvalchemi/models/pme.py [hunk 2]
```diff
@@ -109,6 +109,9 @@ class PMEModelWrapper(nn.Module, BaseModelMixin):
     coulomb_constant : float, optional
         Coulomb prefactor :math:`k_e` in eV·Å/e².
         Defaults to ``14.3996`` (standard value for Å/e/eV unit system).
+    slab_correction : bool, optional
```
Comment (laserkelvin) [inline nvalchemi/models/pme.py:112]: Could you expand this docstring with the slab contract exercised in `test/models/test_slab.py:225-243` and `test/models/test_slab.py:254-273`? Specifically: `data.pbc` is a bool tensor shaped `(B, 3)`, the non-periodic slab axis is `False` such as `[True, True, False]`, fully periodic masks are no-ops, and mixed slab/3D batches are supported.

File: nvalchemi/models/ewald.py [hunk 3]
```diff
@@ -268,6 +273,15 @@ def adapt_input(self, data: AtomicData | Batch, **kwargs: Any) -> dict[str, Any]
                 "(data.cell must be present)."
             )
 
+        if self.slab_correction:
```
Comment (laserkelvin) [inline nvalchemi/models/ewald.py:288]: `adapt_input()` now requires `data.pbc` when slab correction is on, but `input_data()` still omits it. Please add a slab-enabled assertion beside `TestEwaldInputOutput.test_input_data_override` in `test/models/test_ewald.py:201-209` so `EwaldModelWrapper(..., slab_correction=True).input_data()` includes `"pbc"`.

File: nvalchemi/models/pme.py [hunk 4]
```diff
@@ -304,6 +309,15 @@ def adapt_input(self, data: AtomicData | Batch, **kwargs: Any) -> dict[str, Any]
                 "(data.cell must be present)."
             )
 
+        if self.slab_correction:
```
Comment (laserkelvin) [inline nvalchemi/models/pme.py:324]: `adapt_input()` now requires `data.pbc` when slab correction is on, but `input_data()` still omits it. Please add a slab-enabled assertion beside `TestPMEInputOutput.test_input_data_override` in `test/models/test_pme.py:235-243` so `PMEModelWrapper(..., slab_correction=True).input_data()` includes `"pbc"`.

File: nvalchemi/models/ewald.py [hunk 5]
```diff
@@ -385,36 +399,32 @@ def forward(self, data: AtomicData | Batch, **kwargs: Any) -> ModelOutputs:
...
-            cell=cell,
-            k_vectors=k_vectors,
-            alpha=alpha,
-            batch_idx=batch_idx,
-            compute_forces=compute_forces,
-            compute_virial=compute_stresses,
-            hybrid_forces=self.hybrid_forces,
-        )
+        try:
+            result = ewald_summation(
```
Comment (laserkelvin) [inline nvalchemi/models/ewald.py]: The wrapper no longer calls the two functions patched in `test/models/test_ewald.py:557-565` and `test/models/test_ewald.py:589-597`. Please retarget those tests to patch `ewald_summation`, and assert it receives `pbc`/`slab_correction` plus the expected tuple/virial behavior.

File: nvalchemi/models/ewald.py [hunk 6]
```diff
@@ -385,36 +399,32 @@ def forward(self, data: AtomicData | Batch, **kwargs: Any) -> ModelOutputs:
...
+                mask_value=fill_value,
+                batch_idx=batch_idx,
+                compute_forces=compute_forces,
+                compute_virial=compute_stresses,
+                hybrid_forces=self.hybrid_forces,
+                pbc=pbc,
+                slab_correction=self.slab_correction,
+            )
+        except ValueError as exc:
+            if compute_stresses and "not enough values to unpack" in str(exc):
```
Comment (laserkelvin) [inline nvalchemi/models/ewald.py]: `test/models/test_ewald.py:575-600` currently covers the missing-virial path with old split-kernel mocks. Please update that test to exercise this `ewald_summation` path and prefer explicit result-arity validation over matching the text of a lower-level `ValueError`.

File: test/models/test_slab.py [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+
+    compute_neighbors(batch, config=model.model_config.neighbor_config)
+
+
+def _run_model(
+    batch: Batch, method: str, slab_correction: bool
+) -> dict[str, torch.Tensor]:
+    """Evaluate an electrostatics wrapper with energy, forces, and stress."""
+    model = _make_model(method, slab_correction=slab_correction)
+    model.model_config.active_outputs = {"energy", "forces", "stress"}
```
Comment (laserkelvin) [inline test/models/test_slab.py:148]: `_run_model()` enables `"stress"`, but `test_2d_slab_correction_changes_energy` only checks energy (`test/models/test_slab.py:234-243`) and `test_mixed_batch_only_corrects_2d_periodic_system` only checks energy/forces (`test/models/test_slab.py:254-273`). Please add stress assertions for slab changed and 3D system unchanged.

**Review by laserkelvin — COMMENTED** [review]
P2-only review comments

**Review by laserkelvin — APPROVED** [review]

**Comment by laserkelvin** [comment]
/ok to test [523055f](https://github.com/NVIDIA/nvalchemi-toolkit/pull/103/commits/523055f4aeb8731c5284712212533ef49bcacec8)
